docs/src/main/paradox/user/lang/go/eventsourced.md

```diff
@@ -68,26 +78,41 @@ This command handler also validates the command, ensuring the quantity items add
 
 Event handlers are invoked at two points, when restoring entities from the journal, before any commands are handled, and each time a new event is emitted. An event handlers responsibility is to update the state of the entity according to the event. Event handlers are the only place where its safe to mutate the state of the entity at all.
 
-Event handlers are declared by either implementing the `cloudstate.EventHandler` interface
+Event handlers are declared by implementing the `cloudstate.EventHandler` interface.
 
 @@snip [shoppingcart.go](/docs/src/main/paradox/user/lang/go/src/event.go) { #event-handler }
 
-or implementing an unary method that matches the type of the event to be handled. Event handlers are differentiated by the type of event they handle. By default, the type of event an event handler handles will be determined by looking for a single argument that the event handler takes. If for any reason this needs to be overridden, or if the event handler method doesn't exists at all, the event is handed over to the `cloudstate.EventHandler` `Handle` method when the entity implements that interface. The by implementing the `HandleEvent(event interface{}) (handled bool, err error)` method, a event handler indicates if he handled the event or if any occurred, returns an error. The returned error has precedent and the handled flag would not be considered.  
+Emitted events by command handlers get dispatched to the implemented event handler which then decides how to proceed with the event. 
+
+@@snip [shoppingcart.go](/docs/src/main/paradox/user/lang/go/src/shoppingcart.go) { #handle-event } 
 
-Here's an example event handler for the `ItemAdded` event.
+Here's an example of a concrete event handler for the `ItemAdded` event.
 
 @@snip [shoppingcart.go](/docs/src/main/paradox/user/lang/go/src/shoppingcart.go) { #item-added }
 
 ## Producing and handling snapshots
 
-## Multiple behaviors
+Snapshots are an important optimisation for event sourced entities that may contain many events, to ensure that they can be loaded quickly even when they have very long journals. To produce a snapshot, the `cloudstate.Snapshotter` interface has to be implemented that must return a snapshot of the current state in serializable form. 
```

Comment:
I don't think it should be optional to implement this interface, as otherwise by default users might get poor performance and not knowing why. Is that the case?

Reply:
I agree. The text here is a 1:1 from the [documentation in the java support](https://github.com/cloudstateio/cloudstate/blame/705e5b4a8effb8addb505c4146ecababbb341097/docs/src/main/paradox/user/lang/java/eventsourced.md#L81).

Implementation wise we might probably ensure an entity implements this interface otherwise a compilation error would be raised, similar to the `EventHandler` and the `CommandHandler` in the case of this support library.

The [spec says](https://github.com/cloudstateio/cloudstate/blob/master/protocols/protocol/cloudstate/event_sourced.proto#L37):
```
// If present the entity should initialise its state using this snapshot.
    EventSourcedSnapshot snapshot = 3;
```
so far. Though, this should be implemented the same over different support libraries. I have not checked if others raise a runtime check exception or the like.